src/filters/network.rs

```diff
@@ -1466,9 +1502,9 @@ fn check_pattern_hostname_left_right_anchor_filter(
                     // Since it must follow immediatly after the hostname and be a suffix of
                     // the URL, we conclude that filter must be equal to the part of the
                     // url following the hostname.
-                    FilterPart::Simple(f) => get_url_after_hostname(&request.url, hostname) == f,
+                    FilterPart::Simple(f) => get_url_after_hostname(if filter.match_case() {&request.original_url} else {&request.url}, hostname) == f,
                     FilterPart::AnyOf(filters) => {
-                        let url_after_hostname = get_url_after_hostname(&request.url, hostname);
+                        let url_after_hostname = get_url_after_hostname(if filter.match_case() {&request.original_url} else {&request.url}, hostname);
```

Comment:
it'd also help to pull `request_url` out earlier in cases like this too, rather than checking `filter.match_case()` multiple times.

```rust
let request_url = if filter.match_case() {
    &request.original_url
} else {
    &request.url
};
//...
    FilterPart::Simple(f) => get_url_after_hostname(request_url, hostname) == f,
//...
        let url_after_hostname = get_url_after_hostname(request_url, hostname);
```

Reply:
Thank you. Fixed.